spec/workers/new_page_worker_spec.rb

```diff
@@ -0,0 +1,41 @@
+require 'rails_helper'
+require 'sidekiq/testing'
+
+RSpec.describe NewPageWorker do
+  let!(:page) { create(:page_without_content) }
+  let!(:worker) { NewPageWorker.new }
+
+  content_response = Nokogiri::HTML('<html><head><title>Hello World</title></head>
+    <body><h1>Hello from body</h1></body></html>')
+
+  before(:each) do
+    allow_any_instance_of(NewPageWorker).to receive(:download_content).and_return(content_response)
```

Comment:
you already have instance in `:worker`, no need to stub any_instance_of
just stub `allow(worker).to receive().and_return()`